#! /bin/sh /usr/share/dpatch/dpatch-run
## manpage-whatis.dpatch by  <msp@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: https://bugs.launchpad.net/hplip/+bug/285790

@DPATCH@
diff -urNad hplip-2.8.9~/base/utils.py hplip-2.8.9/base/utils.py
--- hplip-2.8.9~/base/utils.py	2008-09-17 03:18:38.000000000 +1000
+++ hplip-2.8.9/base/utils.py	2008-10-19 19:10:42.000000000 +1100
@@ -540,7 +540,7 @@
         w = gtk.Window()
         dialog = gtk.MessageDialog(w, gtk.DIALOG_MODAL | gtk.DIALOG_DESTROY_WITH_PARENT,
                                    gtk.MESSAGE_WARNING, gtk.BUTTONS_OK, 
-                                   "PyQt not installed. GUI not available. Please check that the PyQt package is installed. Exiting.")
+                                   "PyQt not installed. GUI not available. Install \"python-qt3\" with the Synaptic Package Manager (Menu: System -> Administration -> Synaptic Package Manager) or run the command \"sudo apt-get install python-qt3\" in a terminal window.")
         dialog.run()
         dialog.destroy()
 
@@ -1404,7 +1404,8 @@
         log.info("")
 
     elif typ == 'man':
-        log.info('.TH "%s" 1 "%s" Linux "User Manuals"' % (title, version))
+        log.info('.TH "%s" 1 "%s" Linux "User Manuals"' % (crumb, version))
+        log.info(".SH NAME\n%s \- %s" % (crumb,title))
 
         for line in text_list:
             text1, text2, format, trailing_space = line
@@ -1416,10 +1417,10 @@
 
             if format == 'summary':
                 log.info(".SH SYNOPSIS")
-                log.info(".B %s" % text1)
+                log.info(".B %s" % text1.replace('Usage:',''))
 
             elif format == 'name':
-                log.info(".SH NAME\n%s" % text1)
+                log.info(".SH DESCRIPTION\n%s" % text1)
 
             elif format in ('option', 'example', 'note'):
                 if text1:
@@ -1432,6 +1433,25 @@
 
             elif format in ('seealso, para'):
                 log.info(text1)
+   
+        log.info(".SH AUTHOR")
+        log.info("HPLIP (Hewlett-Packard Linux Imaging & Printing) is an")
+        log.info("HP-developed solution for printing, scanning, and faxing with")
+        log.info("HP inkjet and laser based printers in Linux.")
+
+        log.info(".SH REPORTING BUGS")
+        log.info("The HPLIP Launchpad.net site")
+        log.info(".B https://launchpad.net/hplip")
+        log.info("is available to get help, report")
+        log.info("bugs, make suggestions, discuss the HPLIP project or otherwise")
+        log.info("contact the HPLIP Team.")
+
+        log.info(".SH COPYRIGHT")
+        log.info("Copyright (c) 2001-8 Hewlett-Packard Development Company")
+        log.info(".LP")
+        log.info("This software comes with ABSOLUTELY NO WARRANTY.")
+        log.info("This is free software, and you are welcome to distribute it")
+        log.info("under certain conditions. See COPYING file for more details.")
 
         log.info("")
 
